src/index/IndexImpl.cpp

```diff
@@ -1575,7 +1575,11 @@ vector<float> IndexImpl::getMultiplicities(
   std::array<float, 3> m{numTriples / numDistinctSubjects().normal,
                          numTriples / numDistinctPredicates().normal,
                          numTriples / numDistinctObjects().normal};
-  return {m[p.keyOrder()[0]], m[p.keyOrder()[1]], m[p.keyOrder()[2]]};
+  auto [a, b, c, d] = p.keyOrder().keys();
+  // TODO<joka921> could we update this once we have the number of distinct
+  // graphs stored and therefore also a multiplicity for the graph column?
+  AD_CORRECTNESS_CHECK(d == ADDITIONAL_COLUMN_GRAPH_ID);
+  return {m[a], m[b], m[c]};
```

Comment:
This seems to be a recurring pattern. Did you consider exposing an "applyPermutation" function instead of `keys()` that does exactly this on array-like structures with 3/elements?